
mapping: false

use_reflector: true
use_reflector_relocalize: true


reflector_max_length: 0.1
reflector_min_length: 0.03
intensity_min: 500
scan_min_dis: 0.4

scan_topic: scan
odom_topic: odom
map_topic: map

lidar_frame: sensor_Link
base_frame: base_footprint
map_frame: map

# publish frequency
publish_map_freq: 0.25
use_publish_pose_thread: true
publish_pose_freq: 60


# path for saving and loading database
log_path: log/log.txt
err_path: log/err.txt

save_path: log/map/
load_path: log/map/

occupancy_grid_map_resolution: 0.05

# save a keyframe every distance and orientation
min_update_distance: 0.15 # meters
min_update_orientation: 5 # degree

# x y theta

initial_pose: [0,0,0]

preset_lidar_tf: true
lidar_trans_x: 0.857
lidar_trans_y: 0.
lidar_yaw: 0.


# covariances
initial_pose_covariance: [0.0009, 0, 0,
                          0, 0.0009, 0,
                          0, 0, 0.0009]

odom_model_covariance: [0.0004, 0, 0,
                   0, 0.0004, 0,
                   0, 0, 0.0004]


v_covariance_param: 0.1
omega_covariance_param: 0.05
#scan_probability_covariance_param: 1.0
#


scan_probability_covariance_param: 1

reflector_covariance: [0.0015, 0,
                       0, 0.0015]

relocalize_covariance: [0.000625, 0, 0,
                        0, 0.000625, 0,
                        0, 0, 0.0001]

err_cost: 0.15
alert_cost: 0.25
relocalize_min_response: 0.3
scan_buffer_size: 30
loop_scan_search_distance: 10.0
loop_match_min_chain_size: 5
loop_closure_min_response: 0.65
loop_closure_xy_variance_threshold: 0.01
loop_closure_angle_variance_threshold: 0.05
optimize_every_n_constrains: 20

loop_closure_xy_search_range: 3.0
loop_closure_angle_search_range: 20
loop_closure_grid_resolution: 0.05
loop_closure_coarse_xy_search_resolution: 0.1
loop_closure_fine_xy_search_range: 0.1
loop_closure_coarse_angle_search_resolution: 2
loop_closure_fine_angle_search_range: 2
loop_closure_fine_angle_search_resolution: 1

